### ISSUE TITLE
Invalid OS Version check for UINotification availability

### ISSUE THREAD

**Issue by dmiluski** [Appboy/appboy-ios-sdk#128]
+ (NSSet<UNNotificationCategory *> *)getAppboyUNNotificationCategorySet;

#if __IPHONE_OS_VERSION_MIN_REQUIRED < __IPHONE_10_0
+ (NSSet<UIUserNotificationCategory *> *)getAppboyUIUserNotificationCategorySet;
#endif

This min required? Is this mandatory? We have UNNotification migration on the road map, but haven't done this yet? Perhaps, instead of MIN required check, this could just be marked deprecated to match the Foundation API?

Yes... I am ashamed we have not yet updated. But the OS check in this case does not match the OS definition. As UIUserNotifiations are still usable even in iOS11, but Apple prefers UNNotification.

Recommendation:
Instead of OS check, just mark deprecated
+ (NSSet<UIUserNotificationCategory *> *)getAppboyUIUserNotificationCategorySet __deprecated;

* [x] I have checked [Appboy's documentation](https://www.appboy.com/documentation/iOS/) and am still experiencing the issue.

# Report

## Describe your environment.

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / tvos
 Platform Version        | e.g. 8.0
 SDK Version             | e.g. 3.7.6
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once

**Comment by tiffanylu** [comment]
Thanks for the suggestion, @dmiluski! We think moving to a deprecation warning is a good idea and will keep you updated. In the meantime, is the OS version check is currently blocking your ability to integrate the SDK? If so, you can use a category to expose the `getAppboyUIUserNotificationCategorySet` method for immediate use. 

Best,
Tiff

**Comment by edias** [comment]
Do you guys have a solution for the warnings around UNNotificationResponse? 

`UNNotificationResponse' is partial: introduced in iOS 10.0`

We are targeting iOS 10 + but we are still facing lots of warnings on this deprecated API

**Comment by chshapiro** [comment]
Hey @dmiluski and  @edias,

We released version 3.3.0 yesterday which gets rid of all these warnings.
Please let us know if you have any further issues or if we can close out this issue.

Thanks,
Chava

**Comment by edias** [comment]
Hey @chshapiro,

I'm using Appboy-iOS-SDK (3.3.1) and I'm still getting:

`/Appboy-iOS-SDK/AppboyUI/InAppMessage/ABKInAppMessageUIController.m:106:50: 'beforeInAppMessageDisplayed:withKeyboardIsUp:' is deprecated `

**Comment by chshapiro** [comment]
Hey @edias,

That deprecation message is intentional. The delegate method is marked as deprecated, but we are still supporting it to give people a grace period to adopt the new delegates. 

If you still have questions or concerns about the warning message, could you create a separate issue for it? It's not directly related to this issue and we don't want to spam other people who are watching this issue.

Thanks,
Chava

**Comment by edias** [comment]
Hi @chshapiro,

Aside from the one listed above we are still getting

`/Appboy-iOS-SDK/AppboyUI/ABKFeedbackViewController/FeedbackViewController/ABKNavigationFeedbackViewController.m:18:34: Undeclared selector 'setNuiClass:'`

Thanks,
Eduardo Dias

**Comment by chshapiro** [comment]
@edias closing this for now. Please reopen the issue if you have any further questions. Thanks!
